tests/unit/s2n_tls13_pure_shared_secret_test.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "crypto/s2n_pq.h"
+#include "tests/s2n_test.h"
+#include "tests/testlib/s2n_testlib.h"
+#include "tls/s2n_cipher_suites.h"
+#include "tls/s2n_connection.h"
+#include "tls/s2n_tls13_handshake.c"
+#include "utils/s2n_blob.h"
+#include "utils/s2n_mem.h"
+#include "utils/s2n_safety.h"
+
+S2N_RESULT s2n_tls13_derive_secret(struct s2n_connection *conn, s2n_extract_secret_type_t secret_type,
+        s2n_mode mode, struct s2n_blob *secret);
+
+/* PQ shared secret taken from Python-generated KAT */
+#define MLKEM1024_SECRET "B408D5D115713F0A93047DBBEA832E4340787686D59A9A2D106BD662BA0AA035"
+
+/* Expected traffic secrets from Python KAT for MLKEM1024 + AES_128_GCM_SHA256 */
+#define AES_128_MLKEM1024_CLIENT_TRAFFIC_SECRET "8ab94e7f368d1327b624defe057dc31a8f103184f4ac28ff8026a75315a7f8b5"
+#define AES_128_MLKEM1024_SERVER_TRAFFIC_SECRET "204972304f3858ccd147b37b9f0537814276507ee9957e82517133932df49d1d"
+
+/* Expected traffic secrets from Python KAT for MLKEM1024 + AES_256_GCM_SHA384 */
+#define AES_256_MLKEM1024_CLIENT_TRAFFIC_SECRET \
+    "c7b966187dc424ac82656bece5665b0f230b783328f3e38ece38e74768b61f694f3c661214cd9f0a7bece1baeb93f3a0"
+#define AES_256_MLKEM1024_SERVER_TRAFFIC_SECRET \
+    "f920fdabeb2c1b2ca5f016c2124a84f88a18243c24f004bd67a2db238bf49ce02d97259c43ef09e62394a18650dd59ed"
+
+/* Fake transcript string for hashing */
+#define FAKE_TRANSCRIPT "client_hello || server_hello"
+
+struct pure_pq_test_vector {
+    struct s2n_cipher_suite *cipher_suite;
+    const char *transcript;
+    const struct s2n_kem_group *kem_group;
+    struct s2n_blob *pq_secret;
+    struct s2n_blob *expected_client_traffic_secret;
+    struct s2n_blob *expected_server_traffic_secret;
+};
+
+static int set_up_pure_pq_conns(struct s2n_connection *client_conn, struct s2n_connection *server_conn,
+        const struct s2n_kem_group *kem_group, struct s2n_blob *pq_shared_secret);
```

Comment:
You could avoid this declaration by just moving set_up_pure_pq_conns above main. Also, set_up_pure_pq_conns needs to be namespaced with s2n_.